### ISSUE TITLE
Feature request: "brace-style", with selected "1tbs" style, allow "allman" when type definition takes one line

### ISSUE THREAD

**Issue by TokugawaTakesi** [typescript-eslint/typescript-eslint#2208]
I prefer `1tbs`, but below code (subjectively) looks ugly:

```typescript
public get entryPointsGroupsNormalizedSettingsActualForCurrentProjectBuildingMode():
      Map<EntryPointsGroupID, EntryPointsGroupNormalizedSettings> {
    return this.entryPointsGroupsSettings__normalized__actualForCurrentProjectBuildingMode;
  }
```

I suppose exclusively for this case, `allman` style looks better:
```typescript
public get entryPointsGroupsNormalizedSettingsActualForCurrentProjectBuildingMode():
      Map<EntryPointsGroupID, EntryPointsGroupNormalizedSettings> 
{
    return this.entryPointsGroupsSettings__normalized__actualForCurrentProjectBuildingMode;
}
```

- Suggested option name: `allowAllmanForSingleLineType`
- Actual when: function/method name and type definition are on separate lines

**Comment by bradzacher** [comment]
This is where I just prefer to use prettier, as it has standardised and predictable breaking of long lines

```ts
class Foo {
  public get entryPointsGroupsNormalizedSettingsActualForCurrentProjectBuildingMode(): Map<
    EntryPointsGroupID,
    EntryPointsGroupNormalizedSettings
  > {
    return this
      .entryPointsGroupsSettings__normalized__actualForCurrentProjectBuildingMode;
  }

  public get entryPointsGroupsNormalizedSettingsActualForCurrentProjectBuildingMode2(
    argument1: string,
    argument2: boolean
  ): Map<EntryPointsGroupID, EntryPointsGroupNormalizedSettings> {
    return this
      .entryPointsGroupsSettings__normalized__actualForCurrentProjectBuildingMode;
  }
}
```

One manual formatting style for this sort of problem that we use in our Hack codebase at FB is to break at the function's parentheses, even if there's no arguments:

```ts
class Foo {
  public get entryPointsGroupsNormalizedSettingsActualForCurrentProjectBuildingMode(
  ): Map<EntryPointsGroupID, EntryPointsGroupNormalizedSettings> {
    return this
      .entryPointsGroupsSettings__normalized__actualForCurrentProjectBuildingMode;
  }
}
```

---

All of that aside - we won't be looking to add a new configuration style for this (or any) extension rule. Our intention for extension rules is to either:
- align with the base rule, and just add support for TS syntax (and add options for it, if it makes sense based on the base rule)
- enhance the base rule with type information (and add options for it, if it makes sense)

We don't want to break from the base rule to add new features to the rule if they are not specifically for typescript.

In this case - this request is not specific to typescript, so this feature request could be accommodated in the base rule.

Please submit this as an enhancement request to the core ESLint repo.

**Comment by TokugawaTakesi** [comment]
@bradzacher 

> Please submit this as an enhancement request to the core ESLint repo.

O'K, maybe I'll do it, but my suggestion is meaningless without type annotation. Will not I be rejected as "Sorry, we don't support the TypeScript." by ESLint core contributors?

**Comment by bradzacher** [comment]
If you remove the typescript from the equation, then it's a reasonable request:

Instead of

```ts
class Foo {
  get entryPointsGroupsNormalizedSettingsActualForCurrentProjectBuildingMode() {
    return this.entryPointsGroupsSettings__normalized__actualForCurrentProjectBuildingMode;
  }
}
```

You would prefer

```ts
class Foo {
  get entryPointsGroupsNormalizedSettingsActualForCurrentProjectBuildingMode()
  {
    return this.entryPointsGroupsSettings__normalized__actualForCurrentProjectBuildingMode;
  }
}
```
